contracts/interfaces/IWitnetPriceFeed.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.7.0 <0.9.0;
+pragma experimental ABIEncoderV2;
+
+import "../WitnetRequestBoard.sol";
+
+/// @title The Witnet Price Feed basic interface.
+/// @dev Guides implementation of active price feed polling contracts.
+/// @author The Witnet Foundation.
+
+interface IWitnetPriceFeed {
+
+    /// Emitted everytime a new price update request is posted to the Witnet RequestBoard
+    event PriceFeeding(address indexed from, uint256 queryId, uint256 extraFee);
+
+    /// Estimates minimum fee amount in native currency to be paid when 
+    /// requesting a new price update.
+    /// @dev Actual fee depends on the gas price of the `requestUpdate()` transaction.
+    /// @param _gasPrice Gas price expected to be paid when calling `requestUpdate()`
+    function estimateUpdateFee(uint256 _gasPrice) external view returns (uint256);
+
+    /// Returns result of the last valid price update request successfully solved by the Witnet oracle.
+    function lastPrice() external view returns (int256);
+
+    /// Returns the EVM-timestamp when last valid price was reported back from the Witnet oracle.
+    function lastTimestamp() external view returns (uint256);    
+
+    /// Returns tuple containing last valid price and timestamp, as well as status code of latest update
+    /// request that got posted to the Witnet Request Board.
+    /// @return _lastPrice Last valid price reported back from the Witnet oracle.
+    /// @return _lastTimestamp EVM-timestamp of the last valid price.
+    /// @return _latestUpdateStatus Status code of the latest update request.
+    function lastValue() external view returns (
+        int _lastPrice,
+        uint _lastTimestamp,
+        uint _latestUpdateStatus
+    );
+
+    /// Returns identifier of the latest update request posted to the Witnet Request Board.
+    function latestQueryId() external view returns (uint256);
+
+    /// Returns hash of the Witnet Data Request that solved the latest update request.
+    /// @dev Returning 0 while the latest update request remains unsolved.
+    function latestUpdateDrTxHash() external view returns (bytes32);
+
+    /// Returns error message of latest update request posted to the Witnet Request Board.
+    /// @dev Returning empty string if the latest update request remains unsolved, or
+    /// @dev if it was succesfully solved with no errors.
+    function latestUpdateErrorMessage() external view returns (string memory);
```

Comment:
Indeed, `requestUpdate` may front-run `latestUpdateDrTxHash()`. This could happen, for instance, if multiple pricefeed poller instances were configured to attend same pricefeed. The new pricefeed poller and the proposed contract implementations have into account this eventuality (i.e. `requestUpdate` can be called while pending update, as a reward-update mechanism, without reverting). In a vast majority of cases, price update will require just one single state-modifying transaction. 

No matter if a new update request frontruns latest status, `lastValue()` will always return actual last valid price received so far at the very moment in which is called, and actual timestamp in which that price was saved to storage. 

Tests so far prove this approach to reduce operational costs in 20% with respect previous two-phased price update flow.

Reply:
However, data feeds explorer needs to get drtxhash of the data request that solved the last valid price, as to show users a Witnet transaction log of all past price updates. Therefore, it should indeed be returned from `lastValue()`, together with last price and last timestamp.